### PR TITLE
[google_maps_flutter] Adds support for add padding to the map

### PR DESCRIPTION
**Pull request by josesm82**
## Description

This PR allows to set map padding: https://developers.google.com/maps/documentation/android-sdk/map#map_padding

## Related Issues

flutter/flutter#32149

## Checklist

Before you create this PR confirm that it meets all requirements listed below by checking the relevant checkboxes (`[x]`). This will ensure a smooth and quick review process.

- [x] I read the [Contributor Guide] and followed the process outlined there for submitting PRs.
- [ ] My PR includes unit or integration tests for *all* changed/updated/fixed behaviors (See [Contributor Guide]).
- [x] All existing and new tests are passing.
- [x] I updated/added relevant documentation (doc comments with `///`).
- [x] The analyzer (`flutter analyze`) does not report any problems on my PR.
- [x] I read and followed the [Flutter Style Guide].
- [x] The title of the PR starts with the name of the plugin surrounded by square brackets, e.g. [shared_preferences]
- [x] I updated pubspec.yaml with an appropriate new version according to the [pub versioning philosophy].
- [x] I updated CHANGELOG.md to add a description of the change.
- [x] I signed the [CLA].
- [x] I am willing to follow-up on review comments in a timely manner.

## Breaking Change

Does your PR require plugin users to manually update their apps to accommodate your change?

- [ ] Yes, this is a breaking change (please indicate a breaking change in CHANGELOG.md and increment major revision).
- [x] No, this is *not* a breaking change.

[issue database]: https://github.com/flutter/flutter/issues
[Contributor Guide]: https://github.com/flutter/plugins/blob/master/CONTRIBUTING.md
[Flutter Style Guide]: https://github.com/flutter/flutter/wiki/Style-guide-for-Flutter-repo
[pub versioning philosophy]: https://www.dartlang.org/tools/pub/versioning
[CLA]: https://cla.developers.google.com/

### CODE REVIEW

**Comment by iskakaushik** [comment]
Can you add an example for this?

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 1]
```diff
@@ -34,6 +34,7 @@ class GoogleMap extends StatefulWidget {
     this.polygons,
     this.polylines,
     this.circles,
+    this.padding,
```
Comment (DebkanchanSamadder) [inline packages/google_maps_flutter/lib/src/google_map.dart]: ### **Question:** 
Though in the Android and iOS files if padding is null it defaults to 0, shouldn't this line be `this.padding = const EdgeInsets.all(0)` as it is used in `_GoogleMapOptions` by default just like other properties that are used in `_GoogleMapOptions`?

Comment (DebkanchanSamadder) [inline packages/google_maps_flutter/lib/src/google_map.dart]: ### Question

Though in the Android and iOS files if padding is null it defaults to 0, shouldn't this line be this.padding = const EdgeInsets.all(0) as it is used in _GoogleMapOptions by default just like other properties that are used in _GoogleMapOptions?

**Comment by DebkanchanSamadder** [comment]
@iskakaushik Please forgive me for my stupidity as I am new to contributing on GitHub but can I contribute to this PR? I think I can write tests

**Comment by DebkanchanSamadder** [comment]
@iskakaushik Is there a way I can contribute to this PR? I want to write tests and might possibly fix the conflicts. Also I want to open PR but my implementations use this `padding`, is there a way I can open a PR while not conflicting with PR because I just copy pasted this PRs changes to work on my changes

**Comment by iskakaushik** [comment]
@josesm82, do you mind giving @DebkanchanSamadder contribution permissions for this branch on your fork, that way he can add tests and maybe fix the merge conflicts :-)

**Comment by josesm82** [comment]
Done!

**Comment by DebkanchanSamadder** [comment]
@googlebot Author consent granted

**Comment by DebkanchanSamadder** [comment]
@josesm82 your name isn't in `AUTHORS` but from the contributing guide it seems we both have to add our name and email, right @iskakaushik ?

**Comment by DebkanchanSamadder** [comment]
@iskakaushik it looks like I need to give consent, excuse my stupidity but how do I do that?

**Comment by iskakaushik** [comment]
@DebkanchanSamadder looks like you did sign the CLA, i can fix the CLA once the other checks have passed.

**Comment by DebkanchanSamadder** [comment]
@iskakaushik today as soon as I opened vscode dart analyser popped up an error. I don't know anything about iOS development in general but any chance this error is linked to the spa build fails?

### error:


Please review the below report for any information you do not wish to share and report to
  https://github.com/dart-lang/sdk/issues/new

Exception from analysis server (running from VSCode / Dart Code)

### What I was doing

(please describe what you were doing when this exception occurred)

### Request

While responding to request: `analysis.setAnalysisRoots`

### Versions

- Flutter SDK 1.5.4-hotfix.2
- Visual Studio Code 1.35.0
- Dart Code 3.1.0

### Analyzer Info

The analyzer was launched using the arguments:

```text
/Users/debkanchansamadder/flutter/bin/cache/dart-sdk/bin/snapshots/analysis_server.dart.snapshot
--client-id=Dart-Code.dart-code
--client-version=3.1.0
```

### Exception

FileSystemException(path=/Users/debkanchansamadder/Desktop/padding/plugins/packages/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/.symlinks/plugins/google_maps_flutter/example/ios/Pods/GoogleMaps/Maps/Frameworks/GoogleMaps.framework/Resources/GoogleMaps.bundle/GMSCoreResources.bundle/ic_qu_direction_mylocation@; message=Directory listing failed)

```text
#0      _PhysicalFolder.getChildren (package:analyzer/file_system/physical_file_system.dart:334:7)
#1      _OverlayFolder.getChildren (package:analyzer/file_system/overlay_file_system.dart:334:10)
#2      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1144:32)
#3      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#4      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#5      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#6      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#7      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#8      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#9      ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#10     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#11     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#12     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#13     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#14     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#15     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#16     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#17     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#18     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#19     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#20     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#21     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#22     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#23     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#24     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#25     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#26     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#27     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#28     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#29     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#30     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#31     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#32     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#33     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#34     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#35     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#36     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#37     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#38     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#39     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#40     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#41     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#42     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#43     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#44     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#45     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#46     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#47     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#48     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#49     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#50     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#51     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#52     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#53     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#54     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#55     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#56     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#57     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#58     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#59     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#60     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#61     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#62     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#63     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#64     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#65     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#66     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#67     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#68     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#69     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#70     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#71     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#72     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#73     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#74     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#75     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#76     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#77     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#78     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#79     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#80     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#81     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#82     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#83     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#84     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#85     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#86     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#87     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#88     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#89     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#90     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#91     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#92     ContextManagerImpl._createContext.checkManifestFilesIn (package:analysis_server/src/context_manager.dart:1152:33)
#93     ContextManagerImpl._createContext (package:analysis_server/src/context_manager.dart:1158:25)
#94     ContextManagerImpl._createContexts (package:analysis_server/src/context_manager.dart:1191:16)
#95     ContextManagerImpl.setRoots (package:analysis_server/src/context_manager.dart:736:9)
#96     AnalysisServer.setAnalysisRoots (package:analysis_server/src/analysis_server.dart:521:22)
#97     AnalysisDomainHandler.setAnalysisRoots (package:analysis_server/src/domain_analysis.dart:390:14)
#98     AnalysisDomainHandler.handleRequest (package:analysis_server/src/domain_analysis.dart:324:16)
#99     AnalysisServer.handleRequest.<anonymous closure>.<anonymous closure> (package:analysis_server/src/analysis_server.dart:337:45)
#100    _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#101    AnalysisServer.handleRequest.<anonymous closure> (package:analysis_server/src/analysis_server.dart:333:50)
#102    _rootRun (dart:async/zone.dart:1124:13)
#103    _CustomZone.run (dart:async/zone.dart:1021:19)
#104    _runZoned (dart:async/zone.dart:1516:10)
#105    runZoned (dart:async/zone.dart:1500:12)
#106    AnalysisServer.handleRequest (package:analysis_server/src/analysis_server.dart:332:5)
#107    ByteStreamServerChannel._readRequest.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:174:16)
#108    _PerformanceTagImpl.makeCurrentWhile (package:analyzer/src/generated/utilities_general.dart:258:15)
#109    ByteStreamServerChannel._readRequest (package:analysis_server/src/channel/byte_stream_channel.dart:165:47)
#110    ByteStreamServerChannel.listen.<anonymous closure> (package:analysis_server/src/channel/byte_stream_channel.dart:110:26)
#111    _rootRunUnary (dart:async/zone.dart:1132:38)
#112    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#113    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#114    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#115    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#116    _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#117    _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#118    _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#119    _LineSplitterSink._addLines (dart:convert/line_splitter.dart:150:13)
#120    _LineSplitterSink.addSlice (dart:convert/line_splitter.dart:125:5)
#121    StringConversionSinkMixin.add (dart:convert/string_conversion.dart:163:5)
#122    _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#123    _rootRunUnary (dart:async/zone.dart:1132:38)
#124    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#125    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#126    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#127    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#128    _SinkTransformerStreamSubscription._add (dart:async/stream_transformers.dart:68:11)
#129    _EventSinkWrapper.add (dart:async/stream_transformers.dart:15:11)
#130    _StringAdapterSink.add (dart:convert/string_conversion.dart:236:11)
#131    _StringAdapterSink.addSlice (dart:convert/string_conversion.dart:241:7)
#132    _Utf8ConversionSink.addSlice (dart:convert/string_conversion.dart:312:20)
#133    _Utf8ConversionSink.add (dart:convert/string_conversion.dart:305:5)
#134    _ConverterStreamEventSink.add (dart:convert/chunked_conversion.dart:72:18)
#135    _SinkTransformerStreamSubscription._handleData (dart:async/stream_transformers.dart:120:24)
#136    _rootRunUnary (dart:async/zone.dart:1132:38)
#137    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#138    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#139    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#140    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#141    _SyncStreamController._sendData (dart:async/stream_controller.dart:764:19)
#142    _StreamController._add (dart:async/stream_controller.dart:640:7)
#143    _StreamController.add (dart:async/stream_controller.dart:586:5)
#144    _Socket._onData (dart:io-patch/socket_patch.dart:1774:41)
#145    _rootRunUnary (dart:async/zone.dart:1136:13)
#146    _CustomZone.runUnary (dart:async/zone.dart:1029:19)
#147    _CustomZone.runUnaryGuarded (dart:async/zone.dart:931:7)
#148    _BufferingStreamSubscription._sendData (dart:async/stream_impl.dart:336:11)
#149    _BufferingStreamSubscription._add (dart:async/stream_impl.dart:263:7)
#150    _SyncStreamController._sendData (dart:async/stream_controller.dart:764:19)
#151    _StreamController._add (dart:async/stream_controller.dart:640:7)
#152    _StreamController.add (dart:async/stream_controller.dart:586:5)
#153    new _RawSocket.<anonymous closure> (dart:io-patch/socket_patch.dart:1323:33)
#154    _NativeSocket.issueReadEvent.issue (dart:io-patch/socket_patch.dart:844:14)
#155    _microtaskLoop (dart:async/schedule_microtask.dart:41:21)
#156    _startMicrotaskLoop (dart:async/schedule_microtask.dart:50:5)
#157    _runPendingImmediateCallback (dart:isolate-patch/isolate_patch.dart:115:13)
#158    _RawReceivePortImpl._handleMessage (dart:isolate-patch/isolate_patch.dart:172:5)
```

Diagnostics requested after the error occurred are:

```js
[
    {
        "name": "/Users/debkanchansamadder/Desktop/padding/plugins/packages/google_maps_flutter",
        "explicitFileCount": 1,
        "implicitFileCount": -1,
        "workItemQueueLength": 1,
        "cacheEntryExceptions": []
    }
]
```

**Comment by DebkanchanSamadder** [comment]
First, I'm gonna admit I'm pretty stupid. I don't know anything about iOS development.

But as it seems it's failing to build because it can't find some files and cocoapod isn't updated hence it can't run `pod install` as shown in the message

```
Warning: CocoaPods minimum required version 1.6.0 or greater not installed. Skipping pod install.
  CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
  Without CocoaPods, plugins will not work on iOS or macOS.
  For more info, see https://flutter.dev/platform-plugins
To upgrade:
  brew upgrade cocoapods
  pod setup
Running Xcode build...                                          
Xcode build done.                                            4.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    Release.xcconfig line 1: Unable to find included file "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
    Release.xcconfig line 1: Unable to find included file "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    diff: /Podfile.lock: No such file or directory
    diff: /Manifest.lock: No such file or directory
    error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
```

But when I'm running `flutter build iOS --no-codesign` on my Mac it outputs the following
```
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to
device.
Building io.flutter.plugins.googleMobileMapsExample for device (ios-release)...
Running pod install...                                              1.6s
Running Xcode build...                                                  
                                                   
 ├─Building Dart code...                                    31.7s
 ├─Generating dSYM file...                                   0.4s
 ├─Stripping debug symbols...                                0.0s
 ├─Assembling Flutter resources...                           1.8s
 └─Compiling, linking and signing...                         6.7s
Xcode build done.                                           43.8s
Built
/Users/debkanchansamadder/Desktop/padding/plugins/packages/google_maps_flutter/example/build/ios/iphoneos/Runn
er.app.
```

Due to my lack of experience in iOS development I cannot figure out what is happening and what is the problem.

@iskakaushik If you can please tell me what do you think or what can I do.

**Comment by DebkanchanSamadder** [comment]
First, I'm gonna admit I'm pretty stupid. I don't know anything about iOS development.

But as it seems it's failing to build because mainly `cocoapod` isn't updated hence it can't run `pod install` as shown in the message

```
Warning: CocoaPods minimum required version 1.6.0 or greater not installed. Skipping pod install.
  CocoaPods is used to retrieve the iOS and macOS platform side's plugin code that responds to your plugin usage on the Dart side.
  Without CocoaPods, plugins will not work on iOS or macOS.
  For more info, see https://flutter.dev/platform-plugins
To upgrade:
  brew upgrade cocoapods
  pod setup
Running Xcode build...                                          
Xcode build done.                                            4.0s
Failed to build iOS app
Error output from Xcode build:
↳
    ** BUILD FAILED **
Xcode's output:
↳
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    Release.xcconfig line 1: Unable to find included file "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
    Release.xcconfig line 1: Unable to find included file "Pods/Target Support Files/Pods-Runner/Pods-Runner.release.xcconfig"
    === BUILD TARGET Runner OF PROJECT Runner WITH CONFIGURATION Release ===
    diff: /Podfile.lock: No such file or directory
    diff: /Manifest.lock: No such file or directory
    error: The sandbox is not in sync with the Podfile.lock. Run 'pod install' or update your CocoaPods installation.
```

But when I'm running `flutter build iOS --no-codesign` on my Mac it outputs the following
```
Warning: Building for device with codesigning disabled. You will have to manually codesign before deploying to
device.
Building io.flutter.plugins.googleMobileMapsExample for device (ios-release)...
Running pod install...                                              1.6s
Running Xcode build...                                                  
                                                   
 ├─Building Dart code...                                    31.7s
 ├─Generating dSYM file...                                   0.4s
 ├─Stripping debug symbols...                                0.0s
 ├─Assembling Flutter resources...                           1.8s
 └─Compiling, linking and signing...                         6.7s
Xcode build done.                                           43.8s
Built
/Users/debkanchansamadder/Desktop/padding/plugins/packages/google_maps_flutter/example/build/ios/iphoneos/Runn
er.app.
```

Due to my lack of experience in iOS development I cannot definitively say what is happening and what the problem is.

@iskakaushik If you can, please tell me what do you think or what can I do. It also seems not just this but all recent PRs are affected by this.

**Comment by DebkanchanSamadder** [comment]
@iskakaushik Everything looks good. CLA needs to be corrected.

**Comment by DebkanchanSamadder** [comment]
@iskakaushik 

**Tests**: Added ✅
**Conflicts**: Resolved ✅
**Checks**: Queued ⚠️ (Passed in previous commits ✅)
**CLA Consent**: In your hands ✅
**Authors**: Added @josesm82 and myself in `AUTHORS` as Contribution Guide said, BUT the email address attached to commits of @josesm82 are his company's email, it might not be a problem but I thought I should point it out ⚠️

**Comment by DebkanchanSamadder** [comment]
**Review**: Pending⚠️
**Merge**: Pending ⚠️

File: AUTHORS [hunk 2]
```diff
@@ -35,4 +35,5 @@ Thomas Stockx <thomas@stockxit.com>
 Sarbagya Dhaubanjar <sarbagyastha@gmail.com>
 Ozkan Eksi <ozeksi@gmail.com>
 ko2ic <ko2ic.dev@gmail.com>
-Jonathan Younger <jonathan@daikini.com>
```
Comment (iskakaushik) [inline AUTHORS]: Why has this been deleted?

File: packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings [hunk 3]
```diff
@@ -0,0 +1,8 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (iskakaushik) [inline packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: please don't commit your workspace settings.

File: packages/google_maps_flutter/example/lib/padding.dart [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+  }
+
+  Widget _setPaddingButton() {
+    return Padding(
+      padding: const EdgeInsets.all(8.0),
+      child: RaisedButton(
+        child: const Text('Change padding'),
+        onPressed: () async {
+          setState(() {
+            _padding = Random().nextInt(150).toDouble();
```
Comment (iskakaushik) [inline packages/google_maps_flutter/example/lib/padding.dart]: Can you take some kind of a user input here? Perhaps slider with some padding scale? I think that will illustrate this much better.

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 5]
```diff
@@ -85,6 +86,8 @@ class GoogleMap extends StatefulWidget {
   /// Circles to be placed on the map.
   final Set<Circle> circles;
 
+  final EdgeInsets padding;
```
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/google_map.dart]: can you add dart doc explaining this field?

File: packages/google_maps_flutter/lib/src/google_map.dart [hunk 6]
```diff
@@ -34,6 +34,7 @@ class GoogleMap extends StatefulWidget {
     this.polygons,
     this.polylines,
     this.circles,
+    this.padding = const EdgeInsets.all(0),
```
Comment (iskakaushik) [inline packages/google_maps_flutter/lib/src/google_map.dart]: add documentation above explaining this.

File: AUTHORS (hunk 2 above)
Comment (DebkanchanSamadder) [reply AUTHORS]: Accidentally deleted. sorry.

File: packages/google_maps_flutter/lib/src/google_map.dart (hunk 5 above)
Comment (DebkanchanSamadder) [reply packages/google_maps_flutter/lib/src/google_map.dart]: Added `Padding to be set on map. See https://developers.google.com/maps/documentation/android-sdk/map#map_padding for more details.`

File: packages/google_maps_flutter/lib/src/google_map.dart (hunk 6 above)
Comment (DebkanchanSamadder) [reply packages/google_maps_flutter/lib/src/google_map.dart]: Added `If no padding is specified default padding will be 0.`

File: packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings (hunk 3 above)
Comment (DebkanchanSamadder) [reply packages/google_maps_flutter/example/ios/Runner.xcworkspace/xcshareddata/WorkspaceSettings.xcsettings]: Fixed it by adding it to `.gitignore`, `git rm -r --cached . ` and `git add .`

**Comment by DebkanchanSamadder** [comment]
@iskakaushik Looks good?

**Comment by iskakaushik** [comment]
LGTM

File: .gitignore [hunk 7]
```diff
@@ -18,6 +18,7 @@ Pods/
 **/Flutter/flutter_assets/
 ServiceDefinitions.json
 xcuserdata/
+*.xcworkspace
```
Comment (dnfield) [inline .gitignore]: This causes problems.

**Pull request merged**